### PR TITLE
Map 'double' to SQL type 'DOUBLE PRECISION' for an Oracle database (CORE-3165)

### PR DESCRIPTION
**Pull request by maartenc**
## Impact

- [X] Bug fix (non-breaking change which fixes expected existing functionality)
- [ ] Enhancement/New feature (adds functionality without impacting existing logic)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
 
## Description

With this change, a 'double' type will map to the SQL type 'DOUBLE PRECISION' for an Oracle database. 
The current mapping to a FLOAT(24) is not good enough, as described in [CORE-3165](https://liquibase.jira.com/browse/CORE-3165).

### CODE REVIEW

**Comment by nvoxland** [comment]
**Initial/Pre-Review Thoughts**

Thanks for the PR, @maartenc Glad for the help!

Double precision does seem like the correct choice for "double". It will be a change in behavior for oracle users so we'll need to make sure to document it in the release.

Questions I have:
- None

Potential risks:
- None

What could make the full review difficult:
- Nothing

**Comment by filipelautert** [comment]
Hello @maartenc ! Thanks for the code.
I executed the project build and it failed at class DoubleTypeTest - do you mind updating this test file to be according to your changes?

**Comment by maartenc** [comment]
Thanks @filipelautert !
I've updated that DoubleTypeTest as well now :-)

**Review by filipelautert — APPROVED** [review]
Thanks!

File: liquibase-core/src/main/java/liquibase/datatype/core/DoubleType.java [hunk 1]
```diff
@@ -44,7 +44,7 @@ public DatabaseDataType toDatabaseDataType(Database database) {
         }
 
         if (database instanceof OracleDatabase) {
-            return new DatabaseDataType("FLOAT", 24);
+            return new DatabaseDataType("DOUBLE PRECISION");
```
Comment (suryaaki2) [inline liquibase-core/src/main/java/liquibase/datatype/core/DoubleType.java:47]: Can we combine all if conditions for various databases to one single if with || ?

**Review by suryaaki2 — CHANGES_REQUESTED** [review]

File: liquibase-core/src/main/java/liquibase/datatype/core/DoubleType.java (hunk 1 above)
Comment (maartenc) [reply liquibase-core/src/main/java/liquibase/datatype/core/DoubleType.java]: Good suggestion, I pushed an extra change for this

**Review by XDelphiGrl — APPROVED** [review]
PR address an Oracle bug where `FLOAT` was being incorrectly translated to to `DOUBLE` instead of `DOUBLE PRECISION`. 
* Functional and test harness executions passing.
* No additional testing required.

APPROVED

**Pull request merged**